lib/straight-server/order.rb

```diff
@@ -73,31 +73,41 @@ def self.find_by_address(address)
     # couldn't belong to that newer order (their block_height < order's block_height_created_at)
     # will be taken into account.
     def reprocess!
-      raise RuntimeError.new("Order is not in final state") if status < 2
+      raise "Order is not in final state" if status < 2
 
       confirmations_required = gateway.confirmations_required
 
       transactions = gateway.fetch_transactions_for(address)
-      transactions = Straight::Transaction.from_hashes transactions
+      transactions = Straight::Transaction.from_hashes(transactions)
       transactions.select! { |x| x.confirmations >= confirmations_required } if confirmations_required > 0
 
-      same_address_orders = Order.exclude(id: id).where(gateway_id: gateway.id, address: self.address)
+      same_address_orders = self.class.exclude(id: id).where(gateway_id: gateway_id, address: address)
```

Comment:
Not so important here, but explicit reference may conflict with inheritance, and there's an option for it, `gateway.order_class` field: https://github.com/MyceliumGear/straight-server/blob/master/lib/straight-server/gateway.rb#L611